packages/algoliasearch-helper/src/SearchResults/index.js

```diff
@@ -825,6 +840,10 @@ function sortViaFacetOrdering(facetValues, facetOrdering) {
     return acc;
   }, {});
 
+  if (hide.length > 0) {
+    facetValues = removeHiddenFacetValues(facetValues, hide);
+  }
+
   facetValues.forEach(function (item) {
     // hierarchical facets get sorted using their raw name
     var name = item.path || item.name;
```

Comment:
I'm ok with the code you're suggesting, although I think it would be interesting to consider doing the code here and avoiding another iteration:

```suggestion
    var name = item.path || item.name;
    var hidden = hide.indexOf(name) > -1;
    if (!hidden && reverseOrder[name] !== undefined) {
      orderedFacets[reverseOrder[name]] = item;
    } else if (!hidden) {
      remainingFacets.push(item);
    }
```

Not sure if it's worth the difference, but the code doesn't look much harder to me, and should be more efficient in theory.